ДЗ 7.

valenok95 opened the pull request:
перевод репозиториев на spring-data-jpa JpaRepository

Commit:
ДЗ 7.
перевод репозиториев на spring-data-jpa JpaRepository

vlad-iv commented on homework07/src/main/java/com/example/homework07/service/AuthorService.java:
```diff
@@ -11,7 +11,7 @@ public class AuthorService {
 
     public Author getAuthorById(long id) {
-        return authorRepository.getById(id).orElseThrow();
+        return authorRepository.findById(id).orElseThrow();
```
Предлагаю выбросить свое исключение, например можно назвать NotFoundException
в обработчике ошибок для этого исключения вернуть 404 код
Так же во всех методах сервисов

valenok95 replied:
сделал

vlad-iv commented on homework07/src/main/java/com/example/homework07/service/BookService.java:
```diff
@@ -22,35 +22,36 @@ public class BookService {
                 author, genre);
-        Book savedBook = bookRepository.insert(book);
+        Book savedBook = bookRepository.save(book);
         return new BookResponseDto(savedBook.getId(), savedBook.getName(),
```
Предлагаю отделить логику маппинга от бизнес логики в сервисах.
Можно для каждой сущности создать мапперы, например для книг это BookMapper
и минимально 2 метода, toDto и toModel

valenok95 replied:
добавил CommentMapper и BookMapper

vlad-iv commented on homework07/src/main/java/com/example/homework07/service/CommentService.java:
```diff
@@ -22,36 +22,36 @@ public class CommentService {
-        commentRepository.getById(updateComment.getId()).orElseThrow(() -> new NotFoundException(
+        commentRepository.findById(updateComment.getId()).orElseThrow(() -> new NotFoundException(
                 "Комментарий для обновления не обнаружен!"));
         Book book =
```
Предлагаю запретить перенос комментария от одной книги к другой

valenok95 replied:
мы же берём bookId из комментария, как мы можем проставить его другой книге?

vlad-iv commented on homework07/src/main/java/com/example/homework07/service/CommentService.java:
```diff
@@ -22,36 +22,36 @@ public class CommentService {
+        commentRepository.save(comment);
     }
 
     @Transactional
```
real only

valenok95 replied:
готово

vlad-iv commented on homework07/src/main/java/com/example/homework07/service/CommentService.java:
```diff
@@ -22,36 +22,36 @@ public class CommentService {
                 resultComment.getBook().getName(), resultComment.getBook().getAuthor().getName());
     }
 
     public List<CommentResponseDto> getBookCommentsByBookId(long id) {
```
Transactional real only

vlad-iv commented on homework07/src/main/java/com/example/homework07/model/Book.java:
```diff
@@ -1,4 +1,4 @@
-package com.example.homework06.model;
+package com.example.homework07.model;
```
Предлагаю попробовать подход с LAZY загрузкой связей и entity graph для загрузки связи, когда необходимо

valenok95 replied:
добавил

Commit:
ДЗ 7.
перевод репозиториев на spring-data-jpa JpaRepository
обавил маппер, добавил entityGraphs
